tests/test_implements_satisfy_protocols.py

```diff
@@ -0,0 +1,61 @@
+from typing import Protocol, runtime_checkable
```

Comment:
Oooo good callout on this one. I'll use this to validate the other protocols as well